query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -31,6 +32,7 @@ pub async fn create_record(conn: &dyn QueryExt, model: &ModelRef, args: WriteArg
                     let constraint = DatabaseConstraint::Fields(fields.clone());
                     return Err(SqlError::NullConstraintViolation { constraint });
                 }
+                quaint::error::DatabaseConstraint::ForeignKey => unreachable!(),
```

Comment:
This variant is on new versions of quaint, I'll wait for another PR to handle it before merging this one, so it's not implemented twice.